tests/loops/epoch/test_training_epoch_loop.py

```diff
@@ -29,122 +32,213 @@
 _out13 = {"loss": 1.3}
 
 
-@pytest.mark.parametrize(
-    "num_optimizers,batch_outputs,expected",
-    [
-        (1, [], []),
-        (1, [[]], []),
-        # 1 batch
-        (1, [[{0: _out00}]], [_out00]),
-        # 2 batches
-        (1, [[{0: _out00}], [{0: _out01}]], [_out00, _out01]),
-        # 1 batch, 2 optimizers
-        (2, [[{0: _out00, 1: _out01}]], [_out00, _out01]),
-        # 2 batches, 2 optimizers
-        (
-            2,
-            [[{0: _out00, 1: _out01}], [{0: _out10, 1: _out11}]],
-            [[_out00, _out10], [_out01, _out11]],
-        ),
-        # 4 batches, 2 optimizers, different frequency
-        (
-            2,
-            [[{0: _out00}], [{1: _out10}], [{1: _out11}], [{0: _out01}]],
-            [[_out00, _out01], [_out10, _out11]],
-        ),
-        # 1 batch, tbptt with 2 splits (uneven)
-        (1, [[{0: _out00}, {0: _out01}], [{0: _out03}]], [[_out00, _out01], [_out03]]),
-        # 3 batches, tbptt with 2 splits, 2 optimizers alternating
-        (
-            2,
-            [[{0: _out00}, {0: _out01}], [{1: _out10}, {1: _out11}], [{0: _out02}, {0: _out03}]],
-            [[[_out00, _out01], [], [_out02, _out03]], [[], [_out10, _out11], []]],
-        ),
-    ],
-)
-def test_prepare_outputs_training_epoch_end_automatic(num_optimizers, batch_outputs, expected):
-    """Test that the loop converts the nested lists of outputs to the format that the `training_epoch_end` hook
-    currently expects in the case of automatic optimization."""
-    prepared = TrainingEpochLoop._prepare_outputs_training_epoch_end(
-        batch_outputs,
-        automatic=True,
-        num_optimizers=num_optimizers,
+class TestPrepareOutputs:
```

Comment:
I rewrote this into a class to reuse code, but the parametrizations aren't changed.

I kept the tests using the old format, and added new tests for the new one.